dev/test/util.ts

```diff
@@ -33,4 +34,80 @@ describe('isPlainObject()', () => {
     expect(isPlainObject(new Foo())).to.be.false;
     expect(isPlainObject(Object.create(new Foo()))).to.be.false;
   });
+
+  describe('tryGetPreferRestEnvironmentVariable', () => {
+    const sandbox = sinon.createSandbox();
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let warnSpy: any;
+    let originalValue: string | undefined;
+
+    beforeEach(() => {
+      warnSpy = sandbox.spy(console, 'warn');
+      originalValue = process.env.FIRESTORE_PREFER_REST;
+    });
+
+    afterEach(() => {
+      sandbox.restore();
+      if (originalValue === undefined) {
+        delete process.env.FIRESTORE_PREFER_REST;
+      } else {
+        process.env.FIRESTORE_PREFER_REST = originalValue;
+      }
+    });
+
+    it('reads true', async () => {
+      process.env.FIRESTORE_PREFER_REST = 'true';
+      const [prevSet, prevValue] = tryGetPreferRestEnvironmentVariable();
+      expect(prevSet).to.be.true;
+      expect(prevValue).to.be.true;
+    });
+
+    it('reads 1', async () => {
+      process.env.FIRESTORE_PREFER_REST = '1';
+      const [prevSet, prevValue] = tryGetPreferRestEnvironmentVariable();
+      expect(prevSet).to.be.true;
+      expect(prevValue).to.be.true;
+    });
+
+    it('reads false', async () => {
+      process.env.FIRESTORE_PREFER_REST = 'false';
+      const [prevSet, prevValue] = tryGetPreferRestEnvironmentVariable();
+      expect(prevSet).to.be.true;
+      expect(prevValue).to.be.false;
+    });
+
+    it('reads 0', async () => {
+      process.env.FIRESTORE_PREFER_REST = '0';
+      const [prevSet, prevValue] = tryGetPreferRestEnvironmentVariable();
+      expect(prevSet).to.be.true;
+      expect(prevValue).to.be.false;
+    });
+
+    it('ignores case', async () => {
+      process.env.FIRESTORE_PREFER_REST = 'True';
+      const [prevSet, prevValue] = tryGetPreferRestEnvironmentVariable();
+      expect(prevSet).to.be.true;
+      expect(prevValue).to.be.true;
+    });
+
+    it('indicates when the environment variable is not set', async () => {
```

Comment:
"indicates" maybe isn't the right word here. I interpreted it to mean that the test is verifying that there is some "indication" that the environment variable is not set; however, there is no such "indication".

Reply:
let me know if the new test name clarifies it for you. thanks

Reply:
It is perfect. Thank you.